virtual_rainforest/core/data.py

```diff
@@ -0,0 +1,398 @@
+"""API documentation for the :mod:`core.data` module.
+**************************************************
+
+This module handles the population and storage of data sources used to run Virtual
+Rainforest simulations.
+
+The Data class
+==============
+
+The core :class:`~virtual_rainforest.core.data.Data` class is used to store data for the
+variables used in a simulation. It can be used both for data from external sources - for
+example, data used to set the initial environment or time series of inputs - and for
+internal variables used in the simulation. The core of the class is an
+:class:`~xarray.DataSet` object, which is accessible as the
+:attr:`~virtual_rainforest.core.data.Data.data` attribute of the class. This attribute
+can be used directly to access all of the methods of the :class:`~xarray.DataSet` class.
+
+The function of the :class:`~virtual_rainforest.core.data.Data` class itself is to
+automatically validate datasets against the configuration of a simulation before they
+are added to the :attr:`~virtual_rainforest.core.data.Data.data` instance. It also
+stores information about the validation process so that models can confirm that required
+data has been successfully validated.
+
+The core of the :class:`~virtual_rainforest.core.data.Data` class is the
+:class:`~virtual_rainforest.core.data.Data.__setitem__` method. This method provides the
```

Comment:
But these are the API docs? There are separate tutorial pages (`source/virtual_rainforest/core/data.md`, rendered [here](https://virtual-rainforest.readthedocs.io/en/feature-data_system/virtual_rainforest/core/data.html)) that are show how to use it, but the API docs are a little deeper and developer focussed?

Reply:
But remember that not everyone who wants to use your API cares about how it works internally (all right, for now that probably isn't true, but it may be in the future). I just feel like you don't need to spell out which magic methods you're overriding because either a) the reader will already know what `__contains__()` does so you don't need to spell it out or b) they won't, in which it'll just confuse them. Just say something like, "As with a `dict`, you can check if a key is present by doing `'key' in data`" or whatever.

This is a pretty minor stylistic point though, so it doesn't matter much.